### PR TITLE
 feat(connector): [ZSL] add connector template code 

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Zsl only supports automatic payment create and payment status gets updated through webhooks

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: config/config.example.toml [hunk 1]
```diff
@@ -227,6 +227,7 @@ volt.base_url = "https://api.sandbox.volt.io/"
 wise.base_url = "https://api.sandbox.transferwise.tech/"
 worldline.base_url = "https://eu.sandbox.api-ingenico.com/"
 worldpay.base_url = "https://try.access.worldpay.com/"
+zsl.base_url = "https://api.sitoffalb.net/ecp"
```
Comment (SamraatBansal) [inline config/config.example.toml:230]:
```suggestion
zsl.base_url = "https://api.sitoffalb.net/"
```

Comment (SamraatBansal) [reply config/config.example.toml:230]: And subsequently update at all the places.

File: crates/common_enums/src/enums.rs [hunk 2]
```diff
@@ -82,6 +82,7 @@ pub enum AttemptStatus {
 #[strum(serialize_all = "snake_case")]
 /// Connectors eligible for payments routing
 pub enum RoutableConnectors {
+    Zsl,
```
Comment (SamraatBansal) [inline crates/common_enums/src/enums.rs:85]: Place it in alphabetical order

File: crates/router/src/configs/settings.rs [hunk 3]
```diff
@@ -530,6 +530,7 @@ pub struct Connectors {
     pub wise: ConnectorParams,
     pub worldline: ConnectorParams,
     pub worldpay: ConnectorParams,
+    pub zsl: ConnectorParams,
```
Comment (SamraatBansal) [inline crates/router/src/configs/settings.rs:533]: Please maintain the alphabetical order.

File: crates/router/src/core/payments/flows.rs [hunk 4]
```diff
@@ -145,6 +145,7 @@ impl<const T: u8>
 }
 
 default_imp_for_complete_authorize!(
+    connector::Zsl,
```
Comment (SamraatBansal) [inline crates/router/src/core/payments/flows.rs:148]: Please reorder in alphabetical order

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

**Review by SamraatBansal — DISMISSED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by SamraatBansal — APPROVED** [review]

**Pull request merged**